packages/zoe/src/zoe.js

```diff
@@ -147,36 +168,60 @@ import { makeTables } from './state';
  * @typedef {Object} ContractInstance
  * @property {Invite} invite The closely-held administrative invite
  * @property {Object.<string,function>} publicAPI Public functions that can be called on the instance
+ */
+
+/**
+ * @typedef {{}} InstanceHandle - an opaque handle for a contract instance
+ * @typedef {{}} OfferHandle - an opaque handle for an offer
+ * @typedef {{}} InviteHandle - an opaque handle for an invite
+ * @typedef {Object} CustomProperties
  *
- * @typedef {Object} InstanceHandle
- * @typedef {Object} OfferHandle
- * @typedef {Object} InviteHandle
+ * @typedef {object} OfferRecord
+ * @property {OfferHandle} handle - opaque identifier for the offer, used as the key
+ * @property {InstanceHandle} instanceHandle - opaque identifier for the instance
+ * @property {Proposal} proposal - the offer proposal (including want, give, exit)
+ * @property {AmountKeywordRecord} amounts - the amountKeywordRecord that will be turned into payouts
+ *
+ * @typedef {object} InstanceRecord
+ * @property {InstanceHandle} handle - opaque identifier for the instance, used as the table key
+ * @property {InstallationHandle} installationHandle - opaque identifier for the installation
+ * @property {Object.<string,function>} publicAPI - the invite-free publicly accessible API for the contract
+ * @property {object} terms - contract parameters
```

Comment:
Should this be uppercase or lower case?

Reply:
The type system doesn't seem to care. Capitalized for consistency.